### PR TITLE
Hotfix/metrics - for CR puposes

### PR DESCRIPTION
**Pull request by Shahar-Y**
Added appID and AuthType to the metrics, with an option to add more info from claims.
No breaking changes, and tested with kibana locally.

Will not merge. the relevant PR is here: https://github.com/meateam/api-gateway/pull/217
Signed-off-by: shahar-y shya95@gmail.com

### CODE REVIEW

**Comment by yoavrozner** [comment]
Please see [https://github.com/meateam/api-gateway/pull/217#pullrequestreview-608448980](url)

File: server/metrics.go [hunk 1]
```diff
@@ -75,3 +110,40 @@ func initESConfig() ([]es.ClientOptionFunc, string) {
 
 	return elasticOpts, viper.GetString(configElasticsearchIndex)
 }
+
+func extractRequestInfo(c *gin.Context) *extractedTokenInfo {
+	var claims jwt.MapClaims
+	authType := c.GetHeader(oauth.AuthTypeHeader)
+
+	if authType == "" {
+		authType = "NoneAuthType"
```
Comment (yoavrozner) [inline server/metrics.go:119]: I think you should do:
`authType = NoAuthType`
so you will be able to alter only the constant NoAuthType value without the need to alter the "NoneAuthType" string here.

File: oauth/oauth.go [hunk 2]
```diff
@@ -351,7 +352,8 @@ func (m *Middleware) register(ctx *gin.Context, delegator *spb.User) {
 	ctx.Set(user.ContextUserKey, authenticatedUser)
 }
 
-func (m *Middleware) extractTokenFromHeader(ctx *gin.Context) (string, error) {
+// ExtractTokenFromHeader extracts the token from the header and returns it as a string
+func ExtractTokenFromHeader(ctx *gin.Context) (string, error) {
```
Comment (yoavrozner) [inline oauth/oauth.go:356]: Why is this meant to be exported? I don't see another use of it.

File: server/metrics.go [hunk 3]
```diff
@@ -75,3 +110,40 @@ func initESConfig() ([]es.ClientOptionFunc, string) {
...
+	} else {
+		spikeToken, err := oauth.ExtractTokenFromHeader(c)
```
Comment (yoavrozner) [inline server/metrics.go:121]: Don't you have ExtractTokenFromHeader func already? Maybe it relates to:

File: oauth/oauth.go (hunk 2 above)
Comment (Shahar-Y) [reply oauth/oauth.go:356]: It is used in `metrics.go`. I exported it to avoid code duplication

File: server/metrics.go (hunk 3 above)
Comment (Shahar-Y) [reply server/metrics.go:121]: Yup, that's exactly it. I used it to avoid code duplication

File: server/metrics.go (hunk 1 above)
Comment (Shahar-Y) [reply server/metrics.go:119]: Correct, I missed it

File: docker-compose.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-version: '3'
+version: '2.1'
```
[the remark this answers is not in the document]
Comment (Shahar-Y) [reply docker-compose.yml:1]: The version does not work and need fixing

[the remark this answers is not in the document]
Comment (yoavrozner) [reply docker-compose.yml:1]: Please set a date and we will solve this issue once and for all.

File: server/metrics.go [hunk 5]
```diff
@@ -75,3 +110,40 @@ func initESConfig() ([]es.ClientOptionFunc, string) {
...
+	delegatorID := ""
+
+	switch authType {
+	case oauth.DropboxAuthTypeValue:
+		appID = oauth.DropboxAppID
+		delegatorID = c.GetHeader(oauth.AuthUserHeader)
+	case oauth.CargoAuthTypeValue:
+		appID = oauth.CargoAppID
+	case oauth.ServiceAuthCodeTypeValue:
+		appID, _ = claims["clientName"].(string) // get the appID from the claims
```
Comment (yoavrozner) [inline server/metrics.go:144]: "clientName" should be a constant with name that explains the use of it.

File: server/metrics.go (hunk 1 above)
Comment (yoavrozner) [reply server/metrics.go:119]: And didn't change it...

**Review by yoavrozner — CHANGES_REQUESTED** [review]
@Shahar-Y for next time please read and implement the review comments before you ask me to review your PR again.

**Comment by Shahar-Y** [comment]
Closing. The PR is in #223

**Pull request closed without merging**